test_bst.py

```diff
@@ -189,3 +206,17 @@ def test_breadth_first_populated(fixed_eight_node_bst):
     for i in my_tree.breadth_traversal():
         actual.append(i)
     assert expected == actual
+
+
+def test_balancing_heavyleft(fixed_unbalanced_bst):
+    my_tree = fixed_unbalanced_bst
+    assert my_tree.balance() == 2
+    my_tree.re_balance()
+    assert my_tree.balance() == -1
+
+
+def test_balancing_alreadybalanced(fixed_eight_node_bst):
+    my_tree = fixed_eight_node_bst
+    assert my_tree.balance() == 1
+    my_tree.re_balance()
+    assert my_tree.balance() == 1
```

Comment:
These are pretty weak tests.  You can (and should) do better than this.  Explore each of the code paths available, otherwise a small typo in one of them can lead you astray.